test/map/map.test.cpp

```diff
@@ -1684,3 +1687,166 @@ TEST(Map, InvalidUTF8InTile) {
 
     test.runLoop.run();
 }
+
+TEST(Map, ObserveTileLifecycle) {
+    util::RunLoop runLoop;
+
+    struct TileEntry {
+        OverscaledTileID id;
+        std::string sourceID;
+        TileOperation op;
+
+        bool operator==(const TileEntry& other) const noexcept {
```

Comment:
You should be able to default this `operator==` since we use C++20 now.

Reply:
Yup, looks like I don't even need to specify it with Clang 16.